main.nf

```diff
@@ -1,206 +1,175 @@
-        echo "Found test files with common extensions (*.test.js, *.test.py, *.test.java)"
-    
-    else
-        echo "No test files or directories found" >&2
-        echo "No test files found in the repository" > no_tests_found.log
-        exit 1  
+       TESTS_STATUS="PASS"
     fi
+    cd ..
+
+    # Write out a summary status file in CSV format
+    mkdir -p ${out_dir}
+    echo "${repo_name},\${CLONE_STATUS},\${README_STATUS},\${DEP_STATUS},\${TESTS_STATUS}" > status_repo.txt
     """
 }
 
-process CheckAlmanack {
+/*-----------------------------------------------
+   Process: RunAlmanack
+   - Runs the Almanack analysis in a dedicated container.
+   - Copies the repository to /tmp (to avoid slow mounted I/O) and runs Almanack.
+   - Reads the summary from ProcessRepo and appends the Almanack status,
+     producing a uniquely named file (status_almanack_<repo_name>.txt) with columns:
+     ToolName,CloneRepository,CheckReadme,CheckDependencies,CheckTests,Almanack
+-----------------------------------------------*/
+process RunAlmanack {
+    container = 'aditigopalan/cckp-toolkit-almanack:latest'
+    errorStrategy 'ignore'
     input:
-    path repo
-
+      tuple val(repo_url), val(repo_name), path(repo_dir), val(out_dir), path("status_repo.txt")
     output:
-    path "${params.output_dir}/almanack-results.json", emit: almanack_results
-
+      // Emit a tuple: (repo_url, repo_name, out_dir, file(status_almanack_<repo_name>.txt))
+      tuple val(repo_url), val(repo_name), val(out_dir), file("status_almanack_${repo_name}.txt")
     script:
     """
-    mkdir -p ${params.output_dir}  # Create the output directory if it doesn't exist
-    python3 -c "
-    try:
-        import json
-        import almanack
-        result = almanack.table(repo_path='${repo}')
-        print(json.dumps(result, indent=4))
-    except Exception as e:
-        print(f'Error: {e}')
-        exit(1)
-    " > ${params.output_dir}/almanack-results.json 2> ${params.output_dir}/almanack-error.log
+    mkdir -p ${out_dir}
+    # Copy the repository to /tmp for faster I/O
+    cp -r ${repo_dir} /tmp/repo
+    # Run Almanack analysis; if it completes without error, mark as PASS
+    if python3 -c "import json, almanack; print(json.dumps(almanack.table(repo_path='/tmp/repo')))" > ${out_dir}/almanack-results.json 2>/dev/null; then
+         ALMANACK_STATUS="PASS"
+    else
+         ALMANACK_STATUS="FAIL"
+    fi
+    # Append Almanack status to the previous summary line from status_repo.txt and write to a unique file
+    PREV_STATUS=\$(cat status_repo.txt)
+    echo "\${PREV_STATUS},\${ALMANACK_STATUS}" > status_almanack_${repo_name}.txt
     """
 }
 
-process SaveToSynapse {
+/*-----------------------------------------------
+   Process: GenerateReport
+   - Aggregates all uniquely named status_almanack files into one CSV file.
+-----------------------------------------------*/
+process GenerateReport {
```

Comment:
I think this can be covered in a different PR but I think that `collectFile` might be able to replace this entire process.